### PR TITLE
added name support to /enchant command

### PR DESCRIPTION
**Pull request by hYdos**
added support for people to use the command with enchantment names instead of id's.
for example .enchant Sharpness 10 would enchant an item with sharpness 10

### CODE REVIEW

**Comment by Vonr** [comment]
I AGREE WITH THIS PR

**Review by richardletshacks — CHANGES_REQUESTED** [review]
Um this is case-sensitive, please make it case-insensitive

**Comment by hYdos** [comment]
i will work on that now

**Comment by hYdos** [comment]
ill just make it all lowercase i dont think there is a way to make it non case sensitive with maps

**Comment by hYdos** [comment]
i removed case sensitivity

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 1]
```diff
@@ -13,15 +48,31 @@ EnchantCommand::~EnchantCommand()
 
 bool EnchantCommand::execute(std::vector<std::string>* args)
 {
-	assertTrue(args->size() > 1);
+	assertTrue(args->size() > 1); // this line stops the command from been run if the proper args are not given
+
 	int enchantId = 0;
 	int enchantLevel = 32767;
-	if (args->size() > 2)
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp:19]: Please leave it up there

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 2]
```diff
@@ -13,15 +48,31 @@ EnchantCommand::~EnchantCommand()
...
-		enchantLevel = assertInt(args->at(2));
 
-	if (strcmp(args->at(1).c_str(),"all") != 0)
+	try {
+		// convert string to back to lower case
+		std::string data = args->at(1);
+		std::for_each(data.begin(), data.end(), [](char& c) {
+			c = ::tolower(c);
+		});
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp]: `std::transform(data.begin(), data.end(),  data.begin(), ::tolower);`

File: Horion/Command/Commands/EnchantCommand.cpp [hunk 3]
```diff
@@ -1,9 +1,44 @@
...
-
+std::map<std::string, int> enchantMap;
 
 EnchantCommand::EnchantCommand() : IMCCommand("enchant", "Enchants items", "<enchantment> [level]")
 {
+	enchantMap["Protection"] = 0;
+	enchantMap["Fire_Protection"] = 1;
+	enchantMap["Feather_Falling"] = 2;
+	enchantMap["Blast_Protection"] = 3;
+	enchantMap["Projectile_Protection"] = 4;
```
Comment (ghost) [inline Horion/Command/Commands/EnchantCommand.cpp]: Please keep it consistent with uppercase & lowercase

**Review by ghost — CHANGES_REQUESTED** [review]

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 1 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp:19]: sure

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 3 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: fixed

File: Horion/Command/Commands/EnchantCommand.cpp (hunk 2 above)
Comment (hYdos) [reply Horion/Command/Commands/EnchantCommand.cpp]: changed 4 line code to 2 line code

**Comment by realbillie** [comment]
Typo made on line 16. 

enchantMap["aqua_infinity"] = 8;

update to >

enchantMap["aqua_affinity"] = 8;

**Pull request merged**